File: web/catraca/classes/model/Vinculo.php
<?php


class Vinculo{
	
	
	
	private $id;
	private $avulso;
	private $responsavel;
	private $inicioValidade;
	private $finalValidade;
	private $quantidadeDeAlimentosPorTurno;
	private $descricao;
	private $cartao;	

	public function Vinculo(){
		$this->setAvulso(false);
		$this->responsavel = new Usuario();
		$this->cartao= new Cartao();
		
	}
	
	public function setInicioValidade($inicioValidade){
		$this->inicioValidade = $inicioValidade;
	}
	public function getInicioValidade(){
		return $this->inicioValidade;
	}
	public function setFinalValidade($finalValidade){
		$this->finalValidade = $finalValidade;
	}
	public function getFinalValidade(){
		return $this->finalValidade;
	}
	public function setId($id){
		$this->id = $id;
	}
	public function getId(){
		return $this->id;
	}
	public function setAvulso($avulso){
		if($avulso)
			$this->avulso = true;
		else
			$this->avulso = false;
	}
	public function isAvulso(){
		return $this->avulso;
	}
	public function setResponsavel(Usuario $usuario){
		$this->responsavel = $usuario;
	}
	public function getResponsavel(){
		return $this->responsavel;
	}
	public function setQuantidadeDeAlimentosPorTurno($quantidade){
		$this->quantidadeDeAlimentosPorTurno = intval($quantidade);
	}
	public function getQuantidadeDeAlimentosPorTurno(){
		return $this->quantidadeDeAlimentosPorTurno;
	}
	public function setDescricao($descricao){
		$descricao = preg_replace ('/[^a-zA-Z0-9\s]/', '', $descricao);
		$this->descricao = $descricao;
	}
	public function getDescricao(){
		return $this->descricao;
		
	}
	public function setCartao(Cartao $cartao){
		$this->cartao = $cartao;
	}
	public function getCartao(){
		return $this->cartao;
	}
	
	
	
}


?>

File: java/catraca_virtual/catraca_virtual/src/main/java/br/edu/unilab/unicafe/model/Vinculo.java
package br.edu.unilab.unicafe.model;

public class Vinculo {
	private int id;
	private boolean avulso;
	private Usuario responsavel;
	private String inicioValidade;
	private String finalValidade;
	private int quantidadeDeAlimentosPorTurno;
	private String descricao;
	private Cartao cartao;
	private int refeicoesRestantes;
	
	
	public Vinculo(){
		this.cartao = new Cartao();
		this.responsavel = new Usuario();
	}
	
	public int getId() {
		return id;
	}
	public void setId(int id) {
		this.id = id;
	}
	public boolean isAvulso() {
		return avulso;
	}
	public void setAvulso(boolean avulso) {
		this.avulso = avulso;
	}
	public Usuario getResponsavel() {
		return responsavel;
	}
	public void setResponsavel(Usuario responsavel) {
		this.responsavel = responsavel;
	}
	public String getInicioValidade() {
		return inicioValidade;
	}
	public void setInicioValidade(String inicioValidade) {
		this.inicioValidade = inicioValidade;
	}
	public String getFinalValidade() {
		return finalValidade;
	}
	public void setFinalValidade(String finalValidade) {
		this.finalValidade = finalValidade;
	}
	public int getQuantidadeDeAlimentosPorTurno() {
		return quantidadeDeAlimentosPorTurno;
	}
	public void setQuantidadeDeAlimentosPorTurno(int quantidadeDeAlimentosPorTurno) {
		this.quantidadeDeAlimentosPorTurno = quantidadeDeAlimentosPorTurno;
	}
	public String getDescricao() {
		return descricao;
	}
	public void setDescricao(String descricao) {
		this.descricao = descricao;
	}
	public Cartao getCartao() {
		return cartao;
	}
	public void setCartao(Cartao cartao) {
		this.cartao = cartao;
	}
	public int getRefeicoesRestantes() {
		return refeicoesRestantes;
	}
	public void setRefeicoesRestantes(int refeicoesRestantes) {
		this.refeicoesRestantes = refeicoesRestantes;
	}
	
	
}
